src/main/java/org/interledger/cryptoconditions/Condition.java

```diff
@@ -1,56 +1,57 @@
 package org.interledger.cryptoconditions;
 
-import java.net.URI;
-
 /**
- * Java implementation of Crypto-Conditions Condition.
+ * An implementation of a crypto-conditions Condition.
  *
- * @author adrianhopebailie
- * @see<a href="https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/">
- * https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/</a>
+ * @see "https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/"
  */
-public interface Condition {
+public interface Condition extends Comparable<Condition> {
 
   /**
    * The type identifier representing the condition type.
    *
    * @return the type of this condition
    */
-  ConditionType getType();
+  CryptoConditionType getType();
 
   /**
-   * A fingerprint is a binary (aka "octet") string uniquely representing the condition with respect
-   * to other conditions of the same type. Implementations which index conditions MUST use the
-   * entire string or binary encoded condition as the key - not just the fingerprint - as different
-   * conditions of different types may have the same fingerprint.
+   * <p>A fingerprint is a binary (aka "octet") string uniquely representing the condition with
+   * respect to other conditions of the same type. Implementations which index conditions MUST use
+   * the  entire string or binary encoded condition as the key - not just the fingerprint - as
+   * different conditions of different rsa may have the same fingerprint. The length and contents of
+   * the fingerprint are defined by the condition type. The fingerprint is a cryptographically
+   * secure hash of the data which defines the condition, such as a public key.</p>
    *
-   * The length and contents of the fingerprint are defined by the condition type. The fingerprint
-   * is a cryptographically secure hash of the data which defines the condition, such as a public
-   * key.
+   * @return A read-only byte array that contains the unique fingerprint of this condition.
    *
-   * @return the unique fingerprint of this condition
+   * @deprecated This method may be removed in future versions, so callers should use {@link
+   *     #getFingerprintBase64Url()} instead. Java 8 does not currently have the concept of an
+   *     immutable byte array, which means outside callers could manipulate the bytes returned from
+   *     this method and unintentionally (or purposefully) mutate the internal state of this
+   *     Condition.
    */
+  @Deprecated
   byte[] getFingerprint();
```

Comment:
Yeah, I've been going back and forth on this one. The rationale is immutability -- the returned byte-array could be accidentally mutated, but I'm not sure how big of a deal this is. Returning the safe-copy is slower than returning the base64Url string because the base64Url isn't copied every time, but does maintain immutability for the `Condition`.